logstash-core/src/main/java/org/logstash/instrument/metrics/Metric.java

```diff
@@ -88,4 +59,16 @@ default String type() {
         return getType().asString();
     }
 
+    /**
+     * Determine if this metric has a value explicitly set.
+     * @return true if this metric has been set to a specific value, false if it is the default state
+     */
+    boolean isDirty();
```

Comment:
Spoke with @andrewvc on Slack and agreed to leave it in for v1 so that we can make a like-for-like uplift of the serialized output. But we will deprecate the dirty flag immediately so at a soon-to-be later point in time, we can easily remove this flag and explicitly change the behavior.